### ISSUE TITLE
Add silent update install option

### ISSUE THREAD

**Issue by dpetlyuk** [tauri-apps/tauri#1830]
**Is your feature request related to a problem? Please describe.**
We have a self-service kiosk for e-queue, it has no keyboard or mouse, touch screen only. OS is Windows 10. We try to implement auto-update, but it requires administrator elevation, show success dialog window, and don't restart the app.

**Describe the solution you'd like**
Need some setting key to able to silent install updates. Some production cases require auto-update without user interaction, but now someone needs to press OK and restart the app.

**Comment by nothingismagick** [comment]
That's a neat feature and it makes sense in your situation. I guess I see three ways of rolling this out:

Are you using our update mechanism?

**Comment by dpetlyuk** [comment]
Yes, update dialog disabled, updates handled by js.

**Comment by nothingismagick** [comment]
And it still wants an administrator elevation? That's tricky. We may have to wait until @lemarier gets back and has some time to think this over.

**Comment by dpetlyuk** [comment]
Temporary we car run the app as administrator (as I understand it is needed for installation of WebView2, but why it needs on update?). Is it possible now to make the update silent without that (some WiX config)?

**Comment by nothingismagick** [comment]
Well, in the near future WebView2 should be installed already (if you are using a fresh system for your kiosk - which I would expect). Over the next days we will be digging back into windows because we think we can improve startup time there. 

I agree, requiring admin shouldn't be necessary.

**Comment by lemarier** [comment]
> why it needs on update

It’s required because we use wix(msi) and it use  Program Files. (Not sure if Installed in C:\your-app would work?) We could make updater to update the self binary only (no assets/icons etc) I’m leaving for the week end but when i’ll come back I can take a look.

> update silent without that (some WiX config)

It use msiexec.exe I think adding /quiet and /silent do the trick

**Comment by dpetlyuk** [comment]
Thanks, I look forward to it.

**Comment by dpetlyuk** [comment]
Good day! Any news?

**Comment by lemarier** [comment]
Hi @dpetlyuk !

We didn’t forgot you!
It’s added in my current sprint and i’ll try to get to it ASAP.

Its something you need right now?

**Comment by dpetlyuk** [comment]
Hi @lemarier! 

Currently, no new releases are planned, but it would be nice to know the approximate timeframe.

**Comment by lemarier** [comment]
Awesome! I'll try to start working on this probably next week. 
If things go well, you should expect it in ~ 2 patch releases of Tauri to give us time to run tests, so ~ 1 month.

**Comment by euphbriggs** [comment]
I apologize for coming to this discussion late. If you've already worked out a process without elevated privileges, please disregard this message.

I've seen other update processes that require admin rights on Windows avoid the UAC prompt by using the Windows Task Scheduler. During an elevated install process, they create a task that runs with elevated permissions. When the update needs to run, they call this task to execute and avoid the UAC prompt.

If you think this would be helpful, I would be happy to put together a sample.

**Comment by lemarier** [comment]
> I apologize for coming to this discussion late. If you've already worked out a process without elevated privileges, please disregard this message.
> 
> I've seen other update processes that require admin rights on Windows avoid the UAC prompt by using the Windows Task Scheduler. During an elevated install process, they create a task that runs with elevated permissions. When the update needs to run, they call this task to execute and avoid the UAC prompt.
> 
> If you think this would be helpful, I would be happy to put together a sample.

Hi @euphbriggs , 

Thanks a lot for your suggestion, I think it could be useful with an optional configuration as it may cause some security concerns. 

If you want to draft a PR I'll be happy to help/guide you!

Thanks again

**Comment by euphbriggs** [comment]
Sure thing. I'd be happy to take a crack at it.

Should I be looking for the code related to Wix?

**Comment by lemarier** [comment]
Hey @euphbriggs ,

Yes, you can take a look at the [bundler](https://github.com/tauri-apps/tauri/tree/dev/tooling/bundler/src/bundle/windows)

If you need to check the updater process, you can [see it here](https://github.com/tauri-apps/tauri/tree/dev/core/tauri/src/updater)

An [example](https://github.com/tauri-apps/tauri/tree/dev/examples/updater) is configured to kick a new update if you need to test.

Don't hesitate to ask questions, if you are on [discord](https://discord.gg/SpmNs4S) you can ping me as well! `lemarier#1302`

Thanks again

**Comment by euphbriggs** [comment]
@lemarier I put together a very basic prototype of the idea in [this repo](https://github.com/euphbriggs/task_scheduler_prototype).

To keep it simple, it just launches an elevation command prompt, but the concept should work for the update process too.

If this looks like a good direction to you, I'll start looking into how this fits into the Tauri ecosystem and put together a pull request.

**Comment by lemarier** [comment]
hi @euphbriggs,

Well done!

I've tested and it seems promising!

To confirm, the `install_task` should be run from the WIX installer when the user installs the app for the first time.

Then once we got an update, we can elevate with the `/run`

Also, I think we should allow users to enable/disable this feature, so it should probably be in the `tauri.conf.json`

Feel free to draft a PR and if you got any questions, do not hesitate to ping me directly!

Have a great day and thanks again !!!
